### ISSUE TITLE
Starcounter.Response CreateHeadersDictionaryFromHeadersString

### ISSUE THREAD

**Issue by careri** [Starcounter/Starcounter#2939]
There's a bug in the parsing of headers. If one of the headers has no value, only whitespace this code will throw an IndexOutOfRangeException
```
while (Char.IsWhiteSpace(headerAndValue[k]))
    k++
```

I'm guessing that sending an empty header is bad practice, but it shoud work. This is my headers:
```
HSM_CN: innowait.se/HEADSUTV180
HSM_CV: 15.1.175.0
HSM_BOOTSTRAP_HASH: 6E7E285D6DB1285876986760BD8601D9A3807607
HSM_CONFR: 
HSM_Realm: HSM
Host: localhost:8080
```

**Comment by dan31** [comment]
Hi @careri, thanks for the question! @alemoi, please have a check when possible!

**Comment by alemoi** [comment]
Sure, I will check it!

**Comment by alemoi** [comment]
Thank you for reporting! Fixed via 886605a0b7bdaf6b1cfd74071fb2befd811d8aad Available in the following nightly!
